python-api/gstswitch/controller.py

```diff
@@ -394,3 +424,57 @@ def parse_preview_ports(cls, res):
         for tupl in liststr:
             preview_ports.append(int(tupl[0]))
         return preview_ports
+
+    def on_preview_port_added(self, callback):
+        """Register a Callback for the preview_port_added Signal
```

Comment:
I did that for the things I know about, but I don't know why `preview_port_added` transmits `type`, which is a GstCaseType constant (`GST_CASE_*`) -- and I really don't know what a Client should do with that.